askama_shared/src/filters/markdown.rs

```diff
@@ -0,0 +1,17 @@
+use std::borrow::Borrow;
+
+use askama_escape::{Escaper, MarkupDisplay};
+use pulldown_cmark::html::push_html;
+use pulldown_cmark::{Options, Parser};
+
+use crate::error::Result;
+
+pub fn markdown<'a, E, S>(e: E, s: S) -> Result<MarkupDisplay<E, String>>
+where
+    E: Escaper,
+    S: Borrow<&'a str>,
+{
+    let mut buf = String::new();
+    push_html(&mut buf, Parser::new_ext(s.borrow(), Options::all()));
+    Ok(MarkupDisplay::new_safe(buf, e))
```

Comment:
Hmm, how do you reason about safe here? I seem to remember Markdown can include arbitrary HTML.

Reply:
The `Parser` is an `Iterator`, so technically we could escape all [`Event::Html`](https://docs.rs/pulldown-cmark/latest/pulldown_cmark/enum.Event.html#variant.Html). But then we should definitely have an `markdown_unsafe` filter variant

Reply:
Couldn't we just have `|markdown|safe`?

Reply:
But if by default `|markdown` is escaped. Then it also escapes all the HTML generated by `push_html`, right?
E.g. even just `# Title` which results in `<h1>Title</h1>` would then be escaped, or am I missing something?